src/core/display/Container.js

```diff
@@ -606,17 +606,23 @@ Container.prototype.renderCanvas = function (renderer)
 
 /**
  * Destroys the container
- * @param [destroyChildren=false] {boolean} if set to true, all the children will have their destroy method called as well
+ * @param [opts] {object} Options parameter
+ * @param [opts.children=false] {boolean} if set to true, all the children will have their destroy
+ *      method called as well. 'opts' will be passed on to those calls.
+ *
  */
-Container.prototype.destroy = function (destroyChildren)
+Container.prototype.destroy = function (opts)
 {
+    opts = opts || {};
```

Comment:
@marekventur We are not deprecating just providing an alternative (shortened) API for destroy.  No warning necessary. This is going to be a subtle breaking change. Providing a boolean is a convenience that says, "aggressively destroy as much as possible"

Here's what I'm thinking:

```js
// These are equivalent for Sprite.prototype.destroy
sprite.destroy(true);
sprite.destroy({baseTexture: true, texture: true});

// These are equivalent for Text.prototype.destroy
text.destroy(true);
text.destroy({baseTexture: true, texture: true});

// These are equivalent for Container.prototype.destroy
contrainer.destroy(true);
container.destroy({children: true, baseTexture: true, texture: true});
```